File: LayoutTests/http/tests/security/resources/redirect-allow-star.php
<?php
    $url = $_GET["url"];

    $code = isset($_GET["code"]) ? $_GET["code"] : 302;

    header("HTTP/1.1 $code");
    header("Location: $url");
    header("Access-Control-Allow-Origin: *");

    # Workaround for https://bugs.webkit.org/show_bug.cgi?id=77538
    # Caching redirects results in flakiness in tests that dump loader delegates.
    header("Cache-Control: no-store");
?>

File: LayoutTests/http/tests/security/resources/redirect-allow-star.py
#!/usr/bin/env python3

import os
import sys
from urllib.parse import parse_qs

query = parse_qs(os.environ.get('QUERY_STRING', ''), keep_blank_values=True)
url = query.get('url', [''])[0]
code = int(query.get('code', [302])[0])

sys.stdout.write(
    'status: {}\r\n'
    'Location: {}\r\n'
    'Access-Control-Allow-Origin: *\r\n'
    'Cache-Control: no-store\r\n'
    'Content-Type: text/html\r\n\r\n'.format(code, url)
)